File: Src/Visual Studio/SDK/C#/Eruru.ChatRobotRPC For Shared Project/Enums/ChatRobotState.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Eruru.ChatRobotRPC {

	/// <summary>
	/// 在线状态
	/// </summary>
	public enum ChatRobotState {

		/// <summary>
		/// 在线
		/// </summary>
		Online = 1,
		/// <summary>
		/// Q我吧
		/// </summary>
		QMe,
		/// <summary>
		/// 离开
		/// </summary>
		Leave,
		/// <summary>
		/// 忙碌
		/// </summary>
		Busy,
		/// <summary>
		/// 勿扰
		/// </summary>
		DoNotDisturb,
		/// <summary>
		/// 隐身
		/// </summary>
		Invisible

	}

}

File: Src/Intelij IDEA/SDK/Java/Eruru.ChatRobotRPC/src/main/java/org/eruru/chatrobotrpc/enums/ChatRobotState.java
package org.eruru.chatrobotrpc.enums;

/**
 * 在线状态
 */
public enum ChatRobotState {

	/**
	 * 在线
	 */
	online (1),
	/**
	 * Q我吧
	 */
	qMe (2),
	/**
	 * 离开
	 */
	leave (3),
	/**
	 * 忙碌
	 */
	busy (4),
	/**
	 * 勿扰
	 */
	doNotDisturb (5),
	/**
	 * 隐身
	 */
	invisible (6);

	private final int value;

	ChatRobotState (int value) {
		this.value = value;
	}

	/**
	 * 获取枚举对应的整形
	 */
	public int getValue () {
		return value;
	}

	/**
	 * 通过整形获取对应的枚举
	 */
	public static ChatRobotState get (int value) {
		for (ChatRobotState item : ChatRobotState.values ()) {
			if (item.value == value) {
				return item;
			}
		}
		return null;
	}

}